backend/controllers/chatController.js

```diff
@@ -1,18 +1,29 @@
 const mongoose = require("mongoose");
 const { v4: uuidv4 } = require("uuid");
 const UserService = require("../services/UserService");
-
-const { Chat, Message, Poll, User  } = require("../models/index");
+const { Chat, Message, Poll  } = require("../models/index");
+const { create_message } = require("./message_controller.js");
 const path = require('path');
 
 let client_connections = {};
 
 // Implemented by Matthew
 // Include update_clients_in_chat before the function's end
-function create_message(request, response) {
-  const { new_message, chat_id } = request.params;
-  update_clients_in_chat(new_message, chat_id);
-  response.send("User post message");
+async function message_post(request, response) {
+  try {
+    const message_id = await create_message(request, response);
+    const { chat_id } = request.params;
+    await validate_chat_pin(chat_id);
+    
+    const message_document = await Message.findById(message_id)
+      .populate("author", "user_name has_account")
+      .select("author content createdAt"); 
+    update_clients_in_chat(message_document, chat_id);
+    // Explicitly send a response to the request
+    response.status(200).json({ message: "Message created and broadcasted successfully." });
```

Comment:
Remove this part. The client only needs to know that saving the message to the database worked. We can do that from message_controller.